definitions/tools/cnvnator.cwl

```diff
@@ -0,0 +1,92 @@
+#!/usr/bin/env cwl-runner
+
+cwlVersion: v1.0
+class: CommandLineTool
+label: "Run CNVnator to calcualte copy number variations in WGS samples"
+
+arguments: ["source", "/opt/root/bin/thisroot.sh", { shellQuote: false, valueFrom: "&&" },  "/bin/bash", "run_cnvnator.sh"]
+
+requirements:
+    - class: DockerRequirement
+      dockerPull: "mgibio/cnvnator-cwl:0.4"
+    - class: ResourceRequirement
+      ramMin: 20000
+      coresMin: 1
+      tmpdirMin: 10000
+    - class: ShellCommandRequirement
+    - class: InitialWorkDirRequirement
+      listing:
+      - entryname: "run_cnvnator.sh"
+        entry: |
+          #!/bin/bash
+          set -eou pipefail
+
+          # set vars
+          BAM="$1"
+          BIN_SIZE="$2"
+          CHROMOSOMES="${3//,/ }" # input as csv so it's one var to bash script but CNVnator wants them space separated
+          REFERENCE="$4"
+          SAMPLE="$5"
+
+          # create directory to store fasta files. CNVnator wants chromosomes in individual files
+          # while with later versions of CNVnator these steps will accept a fasta.gz file the cnvnator2VCF.pl will not
+          mkdir FASTA_CHRS
+          awk 'BEGIN { CHROM="" } { if ($1~"^>") CHROM=substr($1,2); print $0 > "FASTA_CHRS/"CHROM".fa" }' "$REFERENCE"
```

Comment:
If we're going to run this regularly, should we create a directory of these in advance?  Some of our past reference sequence builds already contain per-chromosome FASTAs (e.g. GRCh37-lite-build37).

Reply:
I've thought about doing something like that. One reason I decided not to was that it would require a new input in the CWL that is either the directory path or a directory object. Or since our input reference is a string check that the directory where that file is located for split chromosomes directory? But that would break if reference is of type `File`.
Figured creating this on the fly would be the easiest way to implement it now. 

In the future changes could be made to the `cnvnator2VCF.pl` script that would allow it to handle a fasta.gz file. Then have the input to this step be a fasta.gz reference which would avoid having to create the individual chromosome files.

Reply:
Makes sense!  Yeah, that would end up requiring a whole new input were we not using a string.